### PR TITLE
betterai-backports / Fix "Granger spit fire extinguisher gives free and infinite evos #2247"

### PR DESCRIPTION
**Pull request by necessarily-equal**
This a backport from bmorel's work.

Fix https://github.com/Unvanquished/Unvanquished/issues/2247.

### CODE REVIEW

**Comment by necessarily-equal** [comment]
LGTM from me, and written by someone else so I guess that I'll merge this in a few days if there's no remarks

File: src/shared/bg_public.h [hunk 1]
```diff
@@ -445,7 +445,8 @@ enum persEnum_t
 #define PS_SPRINTTOGGLE       BIT(3)
 
 // entityState_t->eFlags
-// notice that some flags are overlapped, so their meaning depends on context
+// PLEASE DO NOT MAKE THEM OVERLAP FOR NOTHING:
```
Comment (DolceTriade) [inline src/shared/bg_public.h:448]: What's our policy for overlapping flags? Can you document it here too?

Comment (necessarily-equal) [reply src/shared/bg_public.h:448]: > What's our policy for overlapping flags?

I don't think we have one, but the idea of the commit is that we don't *need* to have one so we better don't, as it avoids possible bugs

Comment (DolceTriade) [reply src/shared/bg_public.h:448]: Yes, but that's not scalable in the long term. As we add more features, we will end up with overlapping flags.

I was hoping we could come up with one right now and encode it as you are touching this file.

My idea for a policy is something like:

Every time you add an overlapping flag you note what it overlaps with and why this is safe.

Comment (necessarily-equal) [reply src/shared/bg_public.h:448]: Crafting a policy sounds good, but also I don't feel like I have a strong opinion on the matter.

Comment (DolceTriade) [reply src/shared/bg_public.h:448]: Then let's add this to the code if you don't care and merge it?

Comment (necessarily-equal) [reply src/shared/bg_public.h:448]: wfm

**Pull request merged**